dot/state/block_test.go

```diff
@@ -520,3 +534,88 @@ func TestAddBlockToBlockTree(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, bs.BestBlockHash(), header.Hash())
 }
+
+func TestNumberIsFinalised(t *testing.T) {
+	bs := newTestBlockState(t, testGenesisHeader)
+	fin, err := bs.NumberIsFinalised(big.NewInt(0))
+	require.NoError(t, err)
+	require.True(t, fin)
+
+	fin, err = bs.NumberIsFinalised(big.NewInt(1))
+	require.NoError(t, err)
+	require.False(t, fin)
+
+	header1 := &types.Header{
+		Number: big.NewInt(1),
+		Digest: types.Digest{
+			types.NewBabeSecondaryPlainPreDigest(0, 1).ToPreRuntimeDigest(),
+		},
+		ParentHash: testGenesisHeader.Hash(),
+	}
+
+	header100 := &types.Header{
+		Number: big.NewInt(100),
+		Digest: types.Digest{
+			types.NewBabeSecondaryPlainPreDigest(0, 100).ToPreRuntimeDigest(),
+		},
+		ParentHash: testGenesisHeader.Hash(),
+	}
+
+	err = bs.SetHeader(header1)
+	require.NoError(t, err)
+	err = bs.db.Put(headerHashKey(header1.Number.Uint64()), header1.Hash().ToBytes())
+	require.NoError(t, err)
+
+	err = bs.SetHeader(header100)
+	require.NoError(t, err)
+	err = bs.SetFinalizedHash(header100.Hash(), 0, 0)
+	require.NoError(t, err)
+
+	fin, err = bs.NumberIsFinalised(big.NewInt(0))
+	require.NoError(t, err)
+	require.True(t, fin)
+
+	fin, err = bs.NumberIsFinalised(big.NewInt(1))
+	require.NoError(t, err)
+	require.True(t, fin)
```

Comment:
🙋‍♂️: wondering why header number 01 is finalized without being called in `bs.SetFinalizedHash`? Is it related to `setFirstSlotOnFinalisation` function?

Reply:
block with number 100 was set as finalized, therefore a block with number 1 is also finalized since there had to be a block 1 that is an ancestor of block 100